vagrant/provisioning/roles/alfresco/files/passencrypt.py

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env python
+import sys
+import hashlib
+passwd = sys.argv[1]
+passwd.encode('utf-16le')
```

Comment:
This line does nothing

Reply:
Why this script is called passencrypt.py when only encode the password and then create a hash from it, and why as a hash algorithm md4 is used, it is considered insecure since 1990, md5 is considered insecure for 20 years now. A better choice as a hash alg would be one of the SHA-2 variants.

Reply:
I agree with the file naming.
Anyway, the MD4 algorithm is used by Alfresco, we cannot change that.

Reply:
@bojmil-armedia , this can be changed. 

With Bcrypt, the hashing algorithm (also called an encoder) can be configured by setting the system.preferred.password.encoding property in the alfresco-global.properties file. The supported values for this property are:

md4
sha256
bcrypt10
If you provide a different value, the repository won’t start.

To maintain backwards compatibility with previous versions, the default setting for this property is:

system.preferred.password.encoding=md4


Here is the link:
https://docs.alfresco.com/content-services/6.1/admin/security/#bcryptoverview

SHA-256 is not best choise but much much better then MD4.  I do not know about bcrypt10, someone can investigate if that is the best option.

Reply:
Wow. I guess, they keep MD4 as default for backward compatibility or something...

Reply:
Let's try with bcrypt10, it should work according to Alfresco docs, and it should be best for new installations.  For current installations, real user passwords are in LDAP, not in Alfresco, so we only have to upgrade admin to the new hashing.

Reply:
The admin password is only in the Alfresco database. We don't need to change anything. Except, if we want to change the password. But that can be done manually.